fearless/Modules/Send/SendPresenter.swift

```diff
@@ -113,14 +115,66 @@ extension SendPresenter: SendViewOutput {
         router.dismiss(view: view)
     }
 
-    func didTapContinueButton() {
-        guard let chainAsset = selectedChainAsset else { return }
-        guard let address = recipientAddress,
-              interactor.validate(address: address, for: chainAsset.chain)
-        else {
-            router.present(message: nil, title: "Incorrect address", closeAction: "Close", from: view)
-            return
+    func validateAddress(with chainAsset: ChainAsset, successCompletion: @escaping (String) -> Void) {
```

Comment:
This construction very cumbersome. Please separate this on the small private methods